### PR TITLE
Get rid of easily removable uses of `const_cast`

### PR DESCRIPTION
**Pull request by rune-scape**
depends on https://github.com/godotengine/godot/pull/94864

gets rid of 138 instances of `const_cast` (greping for `"\Wconst_cast\W"` in all folders except thirdparty gave 282 instances before and 144 after) 

these were the simplest ones i could get rid of. its the reason its in a bulk commit

### CODE REVIEW

**Comment by fire** [comment]
This looks like a great thing. What are the downsides?

**Comment by RandomShaper** [comment]
I haven't checked every case detailedly, but I can say this is reasonable. I mean, if a statement needs `const_cast`, it's usually true that the non-const function needed to write to the object because the relevant data was working as a cache. And in those cases I believe it's better to use `mutable` as a clearer way to signal intent.

File: core/templates/rid_owner.h [hunk 1]
```diff
@@ -163,7 +163,7 @@ class RID_Alloc : public RID_AllocBase {
 		return _allocate_rid();
 	}
 
-	_FORCE_INLINE_ T *get_or_null(const RID &p_rid, bool p_initialize = false) {
+	_FORCE_INLINE_ T *get_or_null(const RID &p_rid, bool p_initialize = false) const {
```
Comment (AThousandShips) [inline core/templates/rid_owner.h:166]: Not sure about this one, it's not a constant function as it does mutate things, so using `const` here is misleading, I'm not sure why it even compiles safely tbh

Many of the owners are marked as `mutable` so I'd leave this alone and make the ones that need this mutable instead

Comment (rune-scape) [reply core/templates/rid_owner.h:166]: the name led me to believe it was like a const function, and it does behave that way if `p_initialize` is false, the default arg, but i understand ill just put mutable in one place

(it compiles bc `chunks` points to mutable memory)

**Comment by adamscott** [comment]
> And in those cases I believe it's better to use `mutable` as a clearer way to signal intent.

I needed to educate myself on the meaning of `mutable`, but it seems way clearer to use this than `const_cast`ing each time. Using `const_cast` screams "this is a hack".

**Review by adamscott — APPROVED** [review]

**Review by huwpascoe — APPROVED** [review]
This looks good.

**Comment by adamscott** [comment]
I asked @rune-scape to rebase the PR. If they aren't available, I'll rebase the PR myself in a few days.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**